### PR TITLE
Add XGlue

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Dataset is ready to merge. An important feature of this dataset is that for each config the train data is in English, while dev and test data are in multiple languages. Therefore, @lhoestq and I decided offline that we will give the dataset the following API, *e.g.* for 

```python
load_dataset("xglue", "ner")  # would give the splits 'train', 'validation.en', 'test.en', 'validation.es', 'test.es', ... 
```
=> therefore one can load a single language test via

```python
load_dataset("xglue", "ner", split="test.es")
```

### CODE REVIEW

File: src/datasets/arrow_reader.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 _SUB_SPEC_RE = re.compile(
     r"""
 ^
- (?P<split>\w+)
+ (?P<split>\w+(-\w+)*)
```
Comment (patrickvonplaten) [inline src/datasets/arrow_reader.py:46]: @lhoestq - I need this to allow splits like `test-de`. Is that fine for you? 
The regex is simply extended to allow for split names that have 0-n hyphens.

Comment (yjernite) [reply src/datasets/arrow_reader.py:46]: @patrickvonplaten I forgot I had the same problem with ELI5. switching to `test_de` should work.

@lhoestq I'm also not a fan of split names with intrinsic meanings. Where is the regular expression documented currently?

**Comment by lhoestq** [comment]
Really cool to add XGlue, this will be a nice addition !

Splits shouldn't depend on the language. There must be configurations for each language, as we're doing for xnli, xtreme, etc.
For example for XGlue we'll have these configurations: NER.de, NER.en etc.

**Review by yjernite — APPROVED** [review]

File: src/datasets/naming.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 _first_cap_re = re.compile("(.)([A-Z][a-z0-9]+)")
 _all_cap_re = re.compile("([a-z0-9])([A-Z])")
 
-_split_re = r"^\w+$"
+_split_re = r"^\w+(\.\w+)*$"
```
Comment (patrickvonplaten) [inline src/datasets/naming.py:26]: Is this fine for you @lhoestq ?

Comment (lhoestq) [reply src/datasets/naming.py:26]: Yes it's fine :)

File: datasets/xglue/README.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
+---
```
Comment (patrickvonplaten) [inline datasets/xglue/README.md:1]: It took me 3h to generate this. The yaml content is complete. The dataset card could be improved.

File: datasets/xglue/xglue.py [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+            citation="",
+            url="",
+        ),
+    ]
+
+    def _info(self):
+        if self.config.name == "ner":
+            features = {
+                "words": datasets.Sequence(datasets.Value("string")),
+                "ner": datasets.Sequence(datasets.Value("string")),
```
Comment (lhoestq) [inline datasets/xglue/xglue.py:355]: Could you use
```
                "tokens": datasets.Sequence(datasets.Value("string")),
                "ner_tags": datasets.Sequence(datasets.ClassLabel(...)),
```
instead ?
For consistency with other NER datasets

File: datasets/xglue/xglue.py [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+            }
+        elif self.config.name == "pos":
+            features = {
+                "words": datasets.Sequence(datasets.Value("string")),
+                "pos": datasets.Sequence(datasets.Value("string")),
```
Comment (lhoestq) [inline datasets/xglue/xglue.py:360]: same with 
```
                "tokens": datasets.Sequence(datasets.Value("string")),
                "pos_tags": datasets.Sequence(datasets.ClassLabel(...)),
```

File: datasets/xglue/xglue.py [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+                "answers": datasets.features.Sequence(
+                    {"answer_start": datasets.Value("int32"), "text": datasets.Value("string")}
+                ),
+                # These are the features of your dataset like images, labels ...
+            }
+        elif self.config.name == "nc":
+            features = {
+                "news_title": datasets.Value("string"),
+                "news_body": datasets.Value("string"),
+                "news_category": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/xglue/xglue.py:375]: is is possible to use ClassLabel here ?

**Review by lhoestq — COMMENTED** [review]
Awesome good job !

I left a few comments about feature naming and types.
Also It looks like the dummy data files are quite heavy (450KB each). It should be possible to remove some data files inside the zip files to make them lighter (for example remove the mlqa files from the dummy_data.zip for for ner)

File: datasets/xglue/README.md [hunk 7]
```diff
@@ -0,0 +1,864 @@
...
+  - crowdsourced
+  - expert-generated
+languages:
+  mlqa:
+  - ar
+  - de
+  - en
+  - es
+  - hi
+	- vi
```
Comment (patrickvonplaten) [inline datasets/xglue/README.md:60]:
```suggestion
  - vi
```

File: datasets/xglue/xglue.py (hunk 6 above)
Comment (patrickvonplaten) [reply datasets/xglue/xglue.py]: > Awesome good job !
> 
> I left a few comments about feature naming and types.
> Also It looks like the dummy data files are quite heavy (450KB each). It should be possible to remove some data files inside the zip files to make them lighter (for example remove the mlqa files from the dummy_data.zip for for ner)

Tried to reduce as much as possible. It total now around ~2MB. Think that's fine given that we have 11 configs a 19 splits

Comment (lhoestq) [reply datasets/xglue/xglue.py]: Thank you so much ! This makes cloning the library faster

File: datasets/xglue/README.md [hunk 8]
```diff
@@ -312,6 +325,522 @@ Therefore, for each config, a cross-lingual pre-trained model should be fine-tun
...
...\u0940 \u092a\u0915\u094d\u0915\u0940 \u0938\u0921\u093c\u0915\u0947\u0902 \u0926\u0915\u094d\u0937\u093f\u0923 \u092e\u0947\u0902 \u092e\u0930\u0915\u0930\u0940 \u0915\u0940 \u0913\u0930 \u0914\u0930 \u092a\u0936\u094d\u091a\u093f\u092e \u092e\u0947\u0902 \u092f\u0941\u0915\u094d\u0915\u093e \u092b\u094d\u0932\u0948\u091f \u0915\u0940 \u0913\u0930 \u091c\u093e\u0924\u0940 \u0939\u0948\u0902\u0964 \u091d\u0940\u0932 \u0938\u0947 \u0909\u0924\u094d\u0924\u0930 \u092a\u0942\u0930\u094d\u0935 \u0915\u0940 \u0913\u0930 \u092c\u0922\u093c\u0924\u0947 \u0939\u0941\u090f \u0935\u094d\u092f\u093e\u092a\u0915 \u0914\u0930 \u0914\u0930 \u0938\u0941\u0935\u094d\u092f\u0935\u0938\u094d\u0925\u093f\u0924 \u0917\u094d\u0930\u0942\u092e \u091d\u0940\u0932 \u0915\u0940 \u0938\u0921\u093c\u0915\u0947\u0902 \u090f\u0915 \u0926\u0930\u094d\u0930\u0947 \u0915\u0947 \u091c\u0930\u093f\u092f\u0947 \u092a\u0947\u091a\u0940\u0926\u093e \u092a\u0939\u093e\u0921\u093c\u093f\u092f\u094b\u0902 \u0938\u0947 \u0939\u094b\u0915\u0930 \u0917\u0941\u091c\u0930\u0924\u0940 \u0939\u0948\u0902\u0964 \u092a\u0939\u0932\u0947 \u0938\u0921\u093c\u0915\u0947\u0902 \u0917\u094d\u0930\u0942\u092e \u0918\u093e\u091f\u0940",
```
Comment (lhoestq) [inline datasets/xglue/README.md:474]: Impressive
not sure if it's possible to show the actual characters

File: datasets/xglue/README.md [hunk 9]
```diff
@@ -312,6 +325,522 @@ Therefore, for each config, a cross-lingual pre-trained model should be fine-tun
...
+
+|   |train |validation.en|validation.de|validation.es|validation.fr|validation.ru|test.en|test.de|test.es|test.fr|test.ru|
+|---|-----:|------------:|------------:|------------:|------------:|------------:|------:|------:|------:|------:|------:|
+|ntg|300000|        10000|        10000|        10000|        10000|        10000|  10000|  10000|  10000|  10000|  10000|
+
+## Dataset Creation
+
+### Curation Rationale
+
+[N/A]
```
Comment (lhoestq) [inline datasets/xglue/README.md:802]:
```suggestion
[More Information Needed]
```
same for the following ones

**Review by lhoestq — APPROVED** [review]
Awesome thanks ! Impressive job

Feel free to merge if you have no other changes to make and CI is green

File: datasets/xglue/README.md (hunk 8 above)
Comment (patrickvonplaten) [reply datasets/xglue/README.md:474]: I'm sure there is a way -> maybe can can use this PR to introduce "First Good Issue" in datasets
